packages/parser-manifest/src/parser.ts

```diff
@@ -0,0 +1,187 @@
+import * as path from 'path';
+import { URL } from 'url';
+
+import {
+    ElementFound,
+    FetchEnd,
+    FetchError,
+    FetchStart,
+    NetworkData,
+    Parser
+} from 'sonarwhal/dist/src/lib/types';
+import { loadJSONFile, normalizeString } from 'sonarwhal/dist/src/lib/utils/misc';
+import {
+    ManifestInvalidJSON,
+    ManifestInvalidSchema,
+    ManifestParsed
+} from './types';
+import { SchemaValidationResult } from 'sonarwhal/dist/src/lib/types/schema-validation-result';
+import { Sonarwhal } from 'sonarwhal/dist/src/lib/sonarwhal';
+import { validate } from 'sonarwhal/dist/src/lib/utils/schema-validator';
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+export default class ManifestParser extends Parser {
+
+    // Event names.
+
+    private fetchEndEventName: string = 'fetch::end::manifest';
+    private fetchErrorEventName: string = 'fetch::error::manifest';
+    private fetchStartEventName: string = 'fetch::start::manifest';
+
+    /* eslint-disable no-invalid-this */
+    private parseEventPrefix: string = 'parse::manifest';
+    private parseEndEventName: string = `${this.parseEventPrefix}::end`;
+    private parseErrorSchemaEventName: string = `${this.parseEventPrefix}::error::schema`;
+    private parseJSONErrorEventName: string = `${this.parseEventPrefix}::error::json`;
+    /* eslint-enable no-invalid-this */
+
+    // Other.
+
+    private schema: any;
+
+    // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+    public constructor(sonarwhal: Sonarwhal) {
+        super(sonarwhal);
+
+        this.schema = loadJSONFile(path.join(__dirname, 'schema.json'));
+
+        sonarwhal.on('element::link', this.fetchManifest.bind(this));
+        sonarwhal.on('fetch::end::manifest', this.validateManifest.bind(this));
+    }
+
+    private async fetchManifest (elementFound: ElementFound) {
+        const { element, resource } = elementFound;
```

Comment:
This currently works because we haven't added the traversing to the local connector but once we do it will break. Maybe add a note or a check to verify we are not analyzing a local resource?

Reply:
Done.